src/docfx/serve/Serve.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class Serve
+    {
+        public static bool Run(CommandLineOptions options)
+        {
+            if (!options.LanguageServer)
+            {
+                Console.WriteLine("Docfx only support serving as a language server in 'Serve' mode, please use `--language-server`");
+            }
+
+            using var stdIn = Console.OpenStandardInput();
```

Comment:
consider move these logic to a `LanguageServer.Start` method. The entire `LanguageServerHost` class could be replaced by a `LanguageServer.Start` method.